developers/bindings/thing-xml.md

```diff
@@ -849,3 +849,84 @@ The full XML schema for Thing type descriptions is specified in the [https://ope
 - The attribute `uri` in the section `config-description` is optional, it _should not_ be specified in bridge/_Thing_/channel type definition files because it's an embedded configuration.
   If the `uri` is _not_ specified, the configuration description is registered as `thing-type:bindingID:id` or `channel-type:bindingID:id` otherwise the given `uri` is used.s
 - If a configuration description is already specified somewhere else and the bridge/_Thing_/channel type wants to (re-)use it, a `config-description-ref` should be used instead.
+
+### Updating Thing Types
+
+Sometimes, when bindings evolve, thing-types need to be modified.
+Since managed things store their structure in a database at the time they are created, only updating the XML is not sufficient.
+Developers can add instructions for the framework to update these things during initialization.
+
+The instructions are provided as XML in the `OH-INF/update` folder.
+The file name can be freely chosen and the file itself can contain instructions for one or more thing-types.
+Instructions for the same thing-type MUST NOT be added in different files.
+
+Update instructions are available for adding, removing or updating channels or channel-groups.
+Changes to configuration parameters don't need update instructions and are performed automatically.
+
+The following update instruction changes the channel-type for the `battery-level` channel to `system:battery-level`.
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<update:update-descriptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                            xmlns:update="https://openhab.org/schemas/update-description/v1.0.0"
+                            xsi:schemaLocation="https://openhab.org/schemas/update-description/v1.0.0 https://openhab.org/schemas/update-description-1.0.0.xsd">
+
+  <thing-type uid="deconz:batterysensor">
+    <instruction-set targetVersion="1">
+      <update-channel id="battery_level">
+        <type>system:battery-level</type>
+      </update-channel>
+    </instruction-set>
+  </thing-type>
+
+</update:update-descriptions>
+```
+
+Different instructions can be combined in one instruction-set.
+The following removes the `water_level` channel from `foo:pool` things and adds a new `chlorine` level
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<update:update-descriptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                            xmlns:update="https://openhab.org/schemas/update-description/v1.0.0"
+                            xsi:schemaLocation="https://openhab.org/schemas/update-description/v1.0.0 https://openhab.org/schemas/update-description-1.0.0.xsd">
+
+  <thing-type uid="foo:pool">
+    <instruction-set targetVersion="1">
+      <remove-channel id="water_level">
+      </remove-channel>
+      <add-channel id="chlorine">
+        <type>foo:concentration</type>
+        <label>Chlorine Concentration</label>
```

Comment:
What is the purpose of the label for `add-channel`?

Reply:
If you use the same channel-type (e.g. `voltage`) for different channels of the thing (e.g. a three-phase power meter) inheriting the label from the channel-type is not desired. By using `label` you can modify the label to be `Voltage L1`, `Voltage L2`, `Voltage L3` .

Reply:
In general (i.e. in the XML that defines the thing), if you define a channel, label and description are taken from the channel-type. If you set label or description in the channel, this will override what is given by the channel-type. The same applies here: explicit label/description take precedence over what is given in the channel-type.

Reply:
So when adding a new channels, is the precedence:
1. `add-channel`/`label` -> `channel`/`label` -> `channel-type`/`label`, or:
2. `add-channel`/`label` -> `channel-type`/`label`

?

If the answer is 1) then I don't see the use-case for overriding it (and doing so without I18N support), because the example you mentioned should already be possible to define correctly in the channel labels, which is also used when creating a new thing or using file-based configuration.

If the answer is 2), maybe this should be fixed in core, so that the label would also be picked from the channel definition without any need to customize this in the update instructions? It seems this would even mean that the label will be in English (for managed things).

Reply:
The label is stored in the JSON database, together with the other channel information. It is put there when the channel is added to the thing, so usually when the thing is created or when the user adds an `extensible` channel. This can't be "fixed" because it is not broken, it works as designed.

When the thing is initially created, then the label is taken from the channel-type, if the XML does not explicitly define another one.

When the channel is added by the update mechanism this is the same: the label is taken from the channel-type, if it is not explicitly set on the update instruction.

Reply:
Okay, thanks for the explanation. So it's option 2) then. Just to be completely sure, does it mean that for:

```xml
<channel-type id="voltage">
	<item-type>Number:Voltage</item-type>
	<label>Voltage</label>
</channel-type>

<channel id="voltage-1" typeId="voltage">
	<label>Voltage L1</label>
</channel>

<add-channel id="voltage-l1">
        <type>foo:voltage</type>
        <label>Voltage L1</label>
</add-channel>
```

With foo.properties:
```
thing-type.foo.bar.channel.voltage-l1.label = Voltage L1
channel-type.foo.voltage.label = Voltage
```

And foo_da.properties:
```
thing-type.foo.bar.channel.voltage-l1.label = Spænding L1
channel-type.foo.voltage.label = Spænding
```

For managed things, the label after the upgrade will be: **Voltage L1**
Whereas for file-based things, or newly created managed things, it will be: **Spænding L1**

And if you omit the label in the update instruction, the result will be:

For managed things, the label after the upgrade will be: **Spænding**
Whereas for file-based things, or newly created managed things, it will be: **Spænding L1**

?

Reply:
i18n is another issue. I didn't check, but if available it has highest priority.

Reply:
Okay, but still it seems like the only way to get fully correct labels and descriptions, is to recreate the thing? Since this PR is about the documentation, I think the only thing that could be improved here would be explaining that:
- If an added channel overrides the channel type label and/or description, these must also be added in the update instructions, since otherwise the default label/description from the channel type will be used.
- Even when doing so, they may not be used, if the channel type label and/or description are translated. _(unconfirmed)_ This does not change the recommendations though, it's still better to have them. This is just explanation. Maybe too detailed.

Reply:
The should not block the PR which is already a huge improvement over having no documentation at all. 🙂 I will probably investigate this a bit more later.